### ISSUE TITLE
GetStdHandle missing from win32.klib

### ISSUE THREAD

**Issue by alanfo** [JetBrains/kotlin-native#781]
I've been trying to do some Windows console programming using Kotlin Native 0.3 on my Windows 10 64 bit laptop. 

I tried initially using the same win32.def file as the sample, namely:
```
headers = windows.h
headerFilter = win**.h
compilerOpts = -DUNICODE -Wno-incompatible-pointer-types -Wno-deprecated-declarations

```
However, when I tried to compile my application, I received an 'unresolved reference' error for the line:
```
val conOut = GetStdHandle(-11)
```
So I checked the win32.kt file generated by cinterop  and, sure enough, the GetStdHandle function was nowhere to be seen despite the fact that it's supposed to be in winbase.h and should therefore be included under the header filter.

I then tried omitting the headerFilter line from win32.def and this time GetStdHandle did appear in win32.kt but unfortunately there was then an error and cinterop didn't get as far as generating win32.klib so I couldn't use it.

Presumably, Mingw-w64 has its own header files which mirror those of the Microsoft compiler but I couldn't actually find them to check that - finding anything on Windows 10 these days is a nightmare!

So, am I doing something wrong here or is there a bug in the cinterop tool?

**Comment by msink** [comment]
Try this:
```
compilerOpts = -DUNICODE -DWINVER=0x0601 -D_WIN32_WINNT=0x0601 -DWINAPI_FAMILY=3 -Wno-incompatible-pointer-types -Wno-deprecated-declarations
```

**Comment by alanfo** [comment]
@msink Thanks for the quick reply and suggestion but, sadly, the results are the same as before.

When I try it with the header filter omitted (and GetStdHandle is then in win32.kt) the error message produced by cinterop is as follows:

```
 1 error generated.
Exception in thread "main" java.lang.Error: Process finished with non-zero exit code: 1
        at org.jetbrains.kotlin.native.interop.gen.jvm.MainKt.runExpectingSuccess(main.kt:119)
        at org.jetbrains.kotlin.native.interop.gen.jvm.MainKt.runCmd(main.kt:175)
        at org.jetbrains.kotlin.native.interop.gen.jvm.MainKt.processLib(main.kt:438)
        at org.jetbrains.kotlin.native.interop.gen.jvm.MainKt.main(main.kt:34)
        at org.jetbrains.kotlin.cli.utilities.MainKt.invokeCinterop(main.kt:30)
        at org.jetbrains.kotlin.cli.utilities.MainKt.main(main.kt:48)
```
though this doesn't appear to have anything to do with the current issue.

The only thing that makes sense here is for GetStdHandle not to be included in winbase.h (or anything else beginning with 'win') under the Mingw distribution. I was wondering whether you were able to find the Mingw header files on your system to check whether that's the case, please?

**Comment by alanfo** [comment]
I wonder actually whether Mingw may use the system header files which, on my machine, are located at:

```C:\Program Files (x86)\Windows Kits\8.0\Include\um``` (there's no 10.0 curiously). There are also some headers in the ```\Include\shared``` folder.

Anyway I've found GetStdHandle in a file called processenv.h which is included in winbase.h so that would explain why it was being filtered out previously.

So, I've managed to build win32.klib with processenv.h included in the .def file and it's worked without any errors. There were a couple of warnings which I've reproduced below in case any one's interested:
```
KtFile: win32.kt
win32-build\kotlin\win32\win32.kt:8860:13: warning: name shadowed: res
        val res = kni_AddFontResourceExA(_name, _fl, _res)
            ^
win32-build\kotlin\win32\win32.kt:8873:13: warning: name shadowed: res
        val res = kni_AddFontResourceExW(_name, _fl, _res)
```
My app is also compiling now so I'll close the issue.

Looks like we're going to have some fun with Windows header files in the future :)
